mkosi/gentoo.py

```diff
@@ -0,0 +1,549 @@
+import multiprocessing
+import os
+import re
+import tarfile
+import urllib.parse
+import urllib.request
+from pathlib import Path
+from textwrap import dedent
+from typing import Dict, List, Sequence
+
+from . import copy_path
+from .backend import (
+    ARG_DEBUG,
+    CommandLineArguments,
+    MkosiException,
+    MkosiPrinter,
+    OutputFormat,
+    die,
+    run_workspace_command,
+)
+
+# https://github.com/python/mypy/issues/5732
+kimg_path: str
+
+
+class Gentoo:
+    arch_profile: Path
+    baselayout_use: Path
+    DEFAULT_NSPAWN_PARAMS: List[str]
+    emerge_default_opts: List[str]
+    arch: str
+    emerge_vars: Dict[str, str]
+    pkgs_boot: List[str]
+    pkgs_sys: List[str]
+    pkgs_fs: List[str]
+    grub_platforms: List[str]
+    UNINSTALL_IGNORE: List[str]
+    root: Path
+    portage_cfg_dir: Path
+    profile_path: Path
+    custom_profile_path: Path
+    ebuild_sh_env_dir: Path
+    dracut_atom = "sys-kernel/dracut"
+
+    EMERGE_UPDATE_OPTS = [
+        "--update",
+        "--tree",
+        "--changed-use",
+        "--newuse",
+        "--deep",
+        "--with-bdeps=y",
+        "--complete-graph-if-new-use=y",
+    ]
+
+    portage_use_flags = [
+        "systemd",  # 'systemd' is a dependancy?
+        "initramfs",
+        "git",  # 'git' for sync-type=git
+        "symlink",  # 'symlink' for kernel
+        "sdl",
+        "-filecaps",
+        "-savedconfig",
+        "-split-bin",
+        "-split-sbin",
+        "-split-usr",
+    ]
+
+    # TODO: portage_features.add("ccache")
+    portage_features = [
+        # -user* are required for access to USER_CONFIG_PATH
+        "-userfetch",
+        "-userpriv",
+        "-usersync",
+        "-usersandbox",
+        "-sandbox",
+        "-pid-sandbox",  # -pid-sandbox is required for cross-compile scenarios
+        "-network-sandbox",
+        "parallel-install",
+        "buildpkg",
+        "binpkg-multi-instance",
+        "-binpkg-docompress",
+        "getbinpkg",
+        "-candy",
+    ]
+
+    ARCHITECTURES = {
```

Comment:
Move this out of the class and into the module proper. This allows you do get rid of `kimg_path`.